### DISCUSSION TITLE
Compress-then-Encrypt

### DISCUSSION THREAD

**Discussion by overheadhunter** [encryption-alliance/unified-vault-format#15] [Rejected]
This is one of the oldest feature requests in Cryptomator (See https://github.com/cryptomator/cryptomator/issues/85).

However, not being able to calculate the cleartext size has been identified as problematic in #11. It also makes random access hard, as we don't know where to start reading/writing.

Nevertheless I felt I should start this discussion. Maybe there are solutions, maybe striking arguments against it. Either should be documented.

**Comment by overheadhunter** [comment]
I could imagine a solution, if ...
* there was a solution for #11, so we could use varying file formats
* we choose a reasonable large block size
* a variable sized file header could contain some kind of index about the uncompressed byte ranges in each encrypted block

**Comment by rfjakob** [comment]
I think this is a security problem, which is why TLS dropped compression some time ago ( https://en.wikipedia.org/wiki/Transport_Layer_Security#CRIME_and_BREACH_attacks )

**Reply by overheadhunter** [reply]
I don't think this is comparable, unless the attacker can trick the user into storing plenty chosen cleartexts in the vault. In this scenario one could argue that the attacker has already gained some form of access on the victim's machine and has easier exploits available.

Regardless I don't believe compression is technically feasible anyway.

**Reply by rfjakob** [reply]
Storing chosen cleartext can be easier than expected, imagine the user has their Thunderbird profile in the vault. Then "storing chosen cleartext" just means sending emails to the victim.

**Reply by overheadhunter** [reply]
fair point
